frontend/app/src/components/helper/NftDetails.vue

```diff
@@ -59,31 +69,33 @@ const props = defineProps({
     required: true,
     type: String
   },
-  styled: { required: false, type: Object, default: () => null }
+  styled: { required: false, type: Object, default: () => null },
+  size: { required: false, type: String, default: '50px' }
 });
 
 const css = useCssModule();
 
 const { identifier } = toRefs(props);
-const { getNftDetails } = useNftAssetInfoStore();
+const { assetInfo } = useAssetInfoRetrieval();
 
-const balanceData = getNftDetails(identifier);
+const balanceData = assetInfo(identifier);
 
 const imageUrl = computed<string | null>(() => {
   return get(balanceData)?.imageUrl ?? '/assets/images/placeholder.svg';
 });
 
-const getCollectionName = (data: NftAsset | null): string | null => {
+const collectionName: ComputedRef<string | null> = computed(() => {
+  const data = get(balanceData);
   if (!data || !data.collectionName) {
     return null;
   }
-  const tokenId = data.identifier.split('_')[3];
+  const tokenId = get(identifier).split('_')[3];
   return `${data.collectionName} #${tokenId}`;
-};
+});
 
-const name = computed<string | null>(() => {
+const name: ComputedRef<string | null> = computed(() => {
   const data = get(balanceData);
-  return data?.name || getCollectionName(data) || null;
+  return data?.name || get(collectionName) || null;
```

Comment:
Do we really need the `|| null`? It seems to me that it will only happen when `get(collectionName)` returns `null`.